### PR TITLE
Move api endpoints to /api/flux/...

### LINKED ISSUE #217: Move api under /api/flux prefix
Will make the service play better with authfe.

And by "play better" I mean not expose prometheus metrics to the world.

### PR DESCRIPTION
**Pull request by philwinder**
Fixes #217
@paulbellamy Is this what you mean?

However, this makes the api endpoints rather verbose.

### CODE REVIEW

**Comment by paulbellamy** [comment]
it is exactly, there'll need to be a matching change in authfe, unfortunately, which makes deployment a bit tricky.

**Comment by paulbellamy** [comment]
Oh will need updating for status stuff merged. Also, is it possible to do the url changes in one place at all? instead of in each route?

**Comment by philwinder** [comment]
Prefix URL now set in one place.

File: http/transport.go [hunk 1]
```diff
@@ -27,8 +27,11 @@ import (
 	"github.com/weaveworks/flux/platform/rpc"
 )
 
+const URL_PREFIX = "/api/flux"
```
Comment (paulbellamy) [inline http/transport.go]: eh, only used in one place, just inline it.

File: http/transport.go [hunk 2]
```diff
@@ -27,8 +27,11 @@ import (
...
+
 func NewRouter() *mux.Router {
 	r := mux.NewRouter()
+	r = r.PathPrefix(URL_PREFIX).Subrouter()
```
Comment (paulbellamy) [inline http/transport.go]: With a subrouter, check that the router lookup route-by-name thing we do in the client still works.

Comment (philwinder) [inline http/transport.go]: I tested manually and it works. But I can't find that in the code. Where is it?

Comment (paulbellamy) [inline http/transport.go]: https://github.com/weaveworks/flux/blob/master/http/transport.go#L759

Comment (philwinder) [inline http/transport.go]: Yeah it works fine.

**Pull request closed without merging**